index.js

```diff
@@ -39,30 +40,100 @@ function getTorrentMetaBuffer () { // TODO Fix parsing bug so this can be remove
 }
 
 function getFile (fpath) {
-  return _getFile(this, fpath)
+  return _getFile(module.exports, fpath)
+  .then(file => {
+    debug('FILE path=' + (file || {}).path, 'found=' + (file != null), file)
+    return file
+  })
+}
+
+function fetch (req, opts) {
+  opts = opts || {}
+  let inject = false
```

Comment:
Inject has a bunch of conditions that would result in it being true. It would be helpful to have a comment that summarizes why inject can be triggered.